sqlx-core/src/postgres/types/ltree.rs

```diff
@@ -0,0 +1,100 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::postgres::{PgArgumentBuffer, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
+use crate::types::Type;
+use std::fmt::{self, Display, Formatter};
+use std::str::FromStr;
+
+
+/// Represents ltree specific errors
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum Error {
+   /// LTree labels can only contain [A-Za-z0-9_]
+    #[error("ltree label cotains invalid characters")]
+    InvalidLtreeLabel,
+
+    /// LTree version not supported
+    #[error("ltree version not supported")]
+    InvalidLtreeVersion,
+}
+
+
+/// Represents an postgres ltree. Not that this is an EXTENSION!
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct PgLTree {
+    labels: Vec<String>,
+}
+
+impl PgLTree {
+    pub fn push(&mut self, label: String) -> Result<(), Error> {
```

Comment:
Now that I think of it there should also be a `new` method, otherwise there's no ergonomic way of building an ltree from scratch.

Reply:
Thats why i added `Default` derive

Reply:
Ah, didn't notice it :sweat_smile:. It would still be worth to have I think